Exploratory Data Analysis  of the dataset 

In [7]:



import pandas as pd

import matplotlib.pyplot as plt
import seaborn as sns 



In [8]:
df=pd.read_csv('covid_and_healthy_spectra.csv')

df.head()

400       402       405       407       410       412       415       417  \
0  0.0 -0.015237 -0.030607 -0.038309 -0.039078 -0.035809 -0.031176 -0.030395   
1  0.0 -0.012098 -0.028164 -0.035189 -0.036138 -0.031050 -0.026015 -0.027539   
2  0.0 -0.013000 -0.029058 -0.035021 -0.034994 -0.033025 -0.028413 -0.028470   
3  0.0 -0.015728 -0.034346 -0.045140 -0.047671 -0.044334 -0.040807 -0.040474   
4  0.0 -0.020355 -0.045839 -0.060556 -0.065805 -0.064988 -0.062097 -0.061955   

        420       422  ...      2101      2103      2104      2105      2107  \
0 -0.033311 -0.031603  ... -0.000553  0.000118  0.000566  0.001245  0.000846   
1 -0.028084 -0.027075  ... -0.000998  0.000278 -0.000123  0.000384  0.001010   
2 -0.029737 -0.029198  ... -0.001554  0.000050  0.000866  0.000877  0.000871   
3 -0.041417 -0.040699  ... -0.001541 -0.000198  0.000202  0.001023  0.000625   
4 -0.064759 -0.066886  ... -0.000088  0.000891  0.000942  0.001294  0.001878   

       2108      2109      2111  2112  diagnostic  
0  0.001106  0.001005  0.000117   0.0     Healthy  
1  0.000583 -0.000397 -0.000160   0.0     Healthy  
2  0.001093  0.001058  0.000614   0.0     Healthy  
3  0.000420  0.000543 -0.000071   0.0     Healthy  
4  0.001739  0.001946  0.001301   0.0     Healthy  

[5 rows x 901 columns]

In [9]:
df.shape

(309, 901)

There are  309 rows and 901 columns in the data . Which may lead to overfitting . So,we need to reduce the number of features to make the prediction task.  
Lasso regresssion seems to be a better option as it reduces number of features. 

In [10]:
#  calculating pearson correlation of features . 

df.corr(method='pearson')

400       402       405       407       410       412       415  \
400   NaN       NaN       NaN       NaN       NaN       NaN       NaN   
402   NaN  1.000000  0.868334  0.743209  0.710465  0.697363  0.725670   
405   NaN  0.868334  1.000000  0.969153  0.950366  0.937134  0.935403   
407   NaN  0.743209  0.969153  1.000000  0.993701  0.982055  0.968134   
410   NaN  0.710465  0.950366  0.993701  1.000000  0.991766  0.975632   
...   ...       ...       ...       ...       ...       ...       ...   
2107  NaN  0.109409  0.014377 -0.047667 -0.053942 -0.031405  0.006487   
2108  NaN  0.093717  0.008923 -0.050834 -0.052125 -0.034914 -0.007962   
2109  NaN  0.075427 -0.007198 -0.063360 -0.066620 -0.048111 -0.020250   
2111  NaN  0.075324 -0.019804 -0.070105 -0.073861 -0.055178 -0.024302   
2112  NaN  0.039602  0.057862  0.065881  0.057573  0.060456  0.064485   

           417       420       422  ...      2100      2101      2103  \
400        NaN       NaN       NaN  ...       NaN       NaN       NaN   
402   0.792408  0.813504  0.807931  ... -0.136898 -0.024881  0.088624   
405   0.955384  0.968244  0.968111  ... -0.091585 -0.044301  0.002989   
407   0.962739  0.966868  0.967240  ... -0.075021 -0.055845 -0.054642   
410   0.964094  0.962466  0.960215  ... -0.069149 -0.059554 -0.059928   
...        ...       ...       ...  ...       ...       ...       ...   
2107  0.038942  0.034891  0.038937  ...  0.582917  0.653254  0.735838   
2108  0.013462  0.018190  0.020346  ...  0.635459  0.675193  0.738895   
2109 -0.002700 -0.006104 -0.009387  ...  0.599741  0.626044  0.656526   
2111 -0.009142 -0.023149 -0.024745  ...  0.523566  0.543517  0.574267   
2112  0.071073  0.065841  0.060482  ...  0.026873  0.047620  0.001733   

          2104      2105      2107      2108      2109      2111      2112  
400        NaN       NaN       NaN       NaN       NaN       NaN       NaN  
402   0.103559  0.074483  0.109409  0.093717  0.075427  0.075324  0.039602  
405   0.017520 -0.042654  0.014377  0.008923 -0.007198 -0.019804  0.057862  
407  -0.043053 -0.106426 -0.047667 -0.050834 -0.063360 -0.070105  0.065881  
410  -0.049824 -0.114140 -0.053942 -0.052125 -0.066620 -0.073861  0.057573  
...        ...       ...       ...       ...       ...       ...       ...  
2107  0.761001  0.870787  1.000000  0.818944  0.654563  0.537270 -0.037246  
2108  0.735510  0.722911  0.818944  1.000000  0.837029  0.581482 -0.012527  
2109  0.645163  0.656798  0.654563  0.837029  1.000000  0.777614  0.043837  
2111  0.600438  0.585064  0.537270  0.581482  0.777614  1.000000  0.029048  
2112  0.045075  0.020566 -0.037246 -0.012527  0.043837  0.029048  1.000000  

[900 rows x 900 columns]